src/transports/PostMessageTransport.ts

```diff
@@ -0,0 +1,88 @@
+import { Transport } from "./Transport";
+import { JSONRPCRequestData, IJSONRPCData } from "../Request";
+
+const openPopup = (url: string) => {
+  const width = 400;
+  const height = window.screen.height;
+  const left = 0;
+  const top = 0;
+
+  return window.open(
+    url,
+    "inspector:popup",
+    `left=${left},top=${top},width=${width},height=${height},resizable,scrollbars=yes,status=1`,
+  );
+};
+
+type PostMessageType = "window" | "iframe";
+
+class PostMessageWindowTransport extends Transport {
+  public uri: string;
+  public frame: undefined | null | Window;
+  public type: PostMessageType;
+  public postMessageID: string;
+
+  constructor(uri: string, type: PostMessageType) {
```

Comment:
Haven't quite groked the whole thing yet, but it seems like this could be more composable if you drop type and simple make type of a function type
```
type PostMessageFrameCreate = (uri: string, postMessageID:string )=> Window | Frame
new PostMessageWindowTransport( uri, openPopup);
new PostMessageWindowTransport(uri,  openIFrame);
new PostMessageWindowTransport(uri, openMemory);
```
Alternatively you could then have the return type interface be a subset of the features required 
for post message transport to work. 
```
type PostMessageFrameCreate = (uri, id) => { somehting that implements eventtlistener on message}
```
I think this will then allow us to have the post message transport supported natively in node, I think then the assumption is that all post message environment specific logic is not in the transport itself 🤔 . In theory one could even evict it entirely from client-js. 

I'm not sure about the usability, it could be we provide the convenience methods for it inside client-js